src/net.c

```diff
@@ -1359,3 +1360,135 @@ void hyper_cleanup_dns(struct hyper_pod *pod)
 
 	close(fd);
 }
+
+int hyper_setup_iptables(struct hyper_pod *pod) {
+	int pid, status;
+	int pipefd[2] = {-1, -1};
+	int pipe_parent_err[2] = {-1, -1};
+	int ret = -1;
+	size_t size, offset = 0;
+	int pipe_size = 0;
+	char *cmd = "iptables-restore";
+
+	if (! pod) {
+		return -1;
+	}
+
+	if (! pod->iptable_rules) {
+		return 0;
+	}
+
+	if (! *(pod->iptable_rules)) {
+		free(pod->iptable_rules);
+		return 0;
+	}
+
+	if (pipe(pipefd) == -1) {
+		fprintf(stderr, "Error while creating pipe for iptables: %s\n",
+				strerror(errno));
+		goto err;
+	}
+
+	if (pipe(pipe_parent_err) == -1) {
+		fprintf(stderr, "Error while creating parent error pipe for "
+				"iptables: %s\n",
+				strerror(errno));
+		goto err;
+	}
+
+	pid = fork();
+	if ( pid == -1) {
+		fprintf(stderr, "Failed to spawn child: %s\n", strerror(errno));
+		goto err;
+	}
+
+	if (pid == 0) {
+		char c;
+		close_if_set(pipefd[1]);
+		close_if_set(pipe_parent_err[1]);
+
+		if (read(pipe_parent_err[0], &c, sizeof(c)) != 0) {
+			fprintf(stderr, "Parent setup failed in iptables-restore\n");
+		}
+
+		/* The rules are sent to the stdin of iptables-restore
+		 */
+		if (dup2(pipefd[0], STDIN_FILENO) == -1) {
+			fprintf(stderr, "Dup call failed : %s\n",
+					strerror(errno));
+			goto err_child;
+		}
+
+		fprintf(stdout, "execing %s\n", cmd);
+		if (execlp(cmd, cmd, "-v", NULL) == -1) {
```

Comment:
This leaks `pipefd[0]` and `pipe_parent_err[0]` to the `exec`'d process (missing `FD_CLOEXEC`?)